unload/logic.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import os
+
+from django.apps import apps
+from django.conf import settings
+from django.template.loaders.app_directories import get_app_template_dirs
+
+
+class BaseSearch(object):
+
+    def get_templates(self, template_dir):
+        """
+        Get full paths to the app's templates.
+
+        :template_dir: full path to the app's template directory
+        :returns: a list of full paths to the app's templates
+        """
+
+        templates = []
+
+        # No template directories were found
+        if not template_dir:
+            return templates
+
+        for dirpath, dirnames, filenames in os.walk(template_dir):
+            for filename in filenames:
+                templates.append(os.path.join(dirpath, filename))
```

Comment:
perhaps you could use `mimetype` module ( [link](https://docs.python.org/2/library/mimetypes.html) ) to only filter `text/*` files, so that you skip binary and other files